tests/simulator_test.cpp

```diff
@@ -37,22 +37,23 @@ class EnergyCollectorTest : public ::testing::Test {
   enum class HitResult { ENERGY_COLLECTORS_EMPTY, NO_HIT, HIT };
 
 protected:
-  std::vector<std::unique_ptr<objects::EnergyCollector>> energyCollectors;
   MockModel nonEmptyModel, emptyModel;
 
   // performs ray hit at energy Collectors by modifying hitData.
   // Returns:
   // ENERGY_COLLECTORS_EMPTY - when no energyCollectors where assigned
   // NO_HIT - when there was no hit
   // HIT - when hit occurred;
-  [[nodiscard]] HitResult performHitCollector(const Ray &ray, float frequency,
-                                              RayHitData *hitData) {
+  [[nodiscard]] HitResult performHitCollector(
+      const Ray &ray,
+      std::vector<std::unique_ptr<objects::EnergyCollector>> *energyCollectors,
```

Comment:
Why do you need to pass a pointer here?

> const std::vector<...>&

Also, I would put collectors as a first param, as this is very rarely to change in your test, unlike Ray.